### PR TITLE
cmake: update minimum version to 3.14.0

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR updates the minimum CMake version required.  This should solve https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1370#issuecomment-602045210

### CODE REVIEW

File: .github/travis/install.sh [hunk 1]
```diff
@@ -3,5 +3,10 @@
 source .github/travis/common.sh
 set -e
 
+# Fix cmake version
```
Comment (mithro) [inline .github/travis/install.sh]: You should use https://apt.kitware.com/

Comment (mithro) [reply .github/travis/install.sh]: https://docs.travis-ci.com/user/languages/cpp/#cmake

Comment (mithro) [reply .github/travis/install.sh]: https://github.com/sourceryinstitute/OpenCoarrays/blob/964360cf183baf07b846ea13818509d14711eba3/.travis.yml#L71-L81

Comment (acomodi) [reply .github/travis/install.sh]: Ok. I am unsure whether I did that correctly. Travis CI still uses 3.12.4.

Would it be ok to install manually cmake in the `install` step following https://apt.kitware.com/ instructions?

Comment (mithro) [reply .github/travis/install.sh]: It looks like it installed cmake 3.17 -- you should add a `which cmake` to see what is going on here....
```
Get:3 https://apt.kitware.com/ubuntu xenial/main amd64 cmake-data all 3.17.0-0kitware1 [1,683 kB]
Get:4 https://apt.kitware.com/ubuntu xenial/main amd64 cmake amd64 3.17.0-0kitware1 [7,422 kB]
```

Comment (acomodi) [reply .github/travis/install.sh]: Turns out that the default cmake that is being used is the one shipped with xenial from Travis is 3.12.4, regardless of the kitware installation.

```
which cmake
/usr/local/cmake-3.12.4/bin/cmake
```

To solve this I have manually extracted the binaries from the cmake official releases

Comment (mithro) [reply .github/travis/install.sh]: How does `/usr/local/cmake-3.12.4/bin/cmake` end up in the path?

Comment (acomodi) [reply .github/travis/install.sh]: Unsure, I believe it gets there as a default when using xenial. Another way probably to solve this issue is to use the apt.kitware package, as previously done, and then update the PATH to use the kitware cmake in the install step.

Comment (mithro) [reply .github/travis/install.sh]: @acomodi Let's use the apt.kitware package and figure out how to make sure the `/usr/local/cmake-3.12.4/bin/cmake` is removed / not used. Worse comes to worse we can do a `sudo rm -rf /usr/local/cmake-3.12.4`

Comment (acomodi) [reply .github/travis/install.sh]: @mithro Done, what was missing was making the priority of the kitware cmake higher in the PATH env var. Travis CI should now turn green.

**Comment by mithro** [comment]
Merge if green.

**Comment by acomodi** [comment]
CI went gree. Merging

**Pull request merged**